test/crc32_test.c

```diff
@@ -0,0 +1,208 @@
+/* crc32_test.c -- crc32 unit test
+ * Copyright (C) 2019-2021 IBM Corporation
+ * Authors: Rogerio Alves    <rogealve@br.ibm.com>
+ *          Matheus Castanho <msc@linux.ibm.com>
+ * For conditions of distribution and use, see copyright notice in zlib.h
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "zbuild.h"
+#ifdef ZLIB_COMPAT
+#  include "zlib.h"
+typedef unsigned long crc32_type;
+typedef unsigned char buf_type;
+typedef unsigned int len_type;
+#  define CRC_FMT "%08lX"
+#else
+#  include "zlib-ng.h"
+typedef uint32_t crc32_type;
+typedef uint8_t buf_type;
```

Comment:
Should be `const uint8_t`.  Assigning string literals to non-const types is deprecated.

Reply:
Fixed.